libs/core/thread_support/src/set_thread_name.cpp

```diff
@@ -99,7 +99,7 @@ namespace hpx::util { namespace detail {
     }
 }}    // namespace hpx::util::detail
 
-#elif defined(__NetBSD__) && defined(HPX_HAVE_NAMEABLE_THREADS)
+#elif defined(__NetBSD__) && defined(HPX_HAVE_PTHREAD_SETNAME_NP)
```

Comment:
The code here is duplicated 3 times. Can we unify that? I also think that we need to check for `HPX_HAVE_PTHREAD_SETNAME_NP` only. There is no need to test for the platform anymore. As long as this function exists we're fine and we don't care about the platform.